tests/playbooks/tests_bridge.yml

```diff
@@ -2,19 +2,17 @@
 ---
 - name: Test configuring bridges
   hosts: all
-  vars:
-    interface: LSR-TST-br31
-
   tasks:
-    - name: "set interface={{ interface }}"
-      set_fact:
-        interface: "{{ interface }}"
-    - include_tasks: tasks/show_interfaces.yml
-    - include_tasks: tasks/assert_device_absent.yml
+    - block:
+        - include_tasks: tasks/show_interfaces.yml
+        - include_tasks: tasks/assert_device_absent.yml
+      vars:
+        interface: LSR-TST-br31
 
 - name: Add test bridge
   hosts: all
   vars:
+    interface: LSR-TST-br31
```

Comment:
Repeating the interface name again and again defeats the purpose of having a variable. Can you please explain why this is needed to run in one VM? To me this seems to be needed only if all playbooks are included in one big playbook. All the playbooks can also run independently in one VM, then the facts of each playbook do not influence each other.

Reply:
> Repeating the interface name again and again defeats the purpose of having a variable. Can you please explain why this is needed to run in one VM? To me this seems to be needed only if all playbooks are included in one big playbook.

Correct.  One of the ways we are looking to improve the performance of testing is to run several test playbooks sequentially on the same machine, rather than having to start a VM for each test playbook.

> All the playbooks can also run independently in one VM, then the facts of each playbook do not influence each other.

Not sure what you mean.

Reply:
Note: Without this pr, running this pair of tests on one VM, the second test fails as in the output - [0].
tox -e qemu-ansible-core-2.12 -- --image-name rhel-8 tests/tests_bridge_nm.yml tests/tests_change_indication_on_repeat_run.yml

[0]
```
TASK [Create veth interface LSR-TST-br31] **************************************
task path: /home/nhosoi/linux-system-roles/network/tests/tasks/manage_test_interface.yml:21
Friday 11 March 2022  10:10:14 -0800 (0:00:10.731)       0:01:02.450 ********** 
failed: [/home/nhosoi/.cache/linux-system-roles/rhel-8.qcow2] (item=ip link add LSR-TST-br31 type veth pee
r name peerLSR-TST-br31) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": [
        "ip",
        "link",
        "add",
        "LSR-TST-br31",
        "type",
        "veth",
        "peer",
        "name",
        "peerLSR-TST-br31"
    ],
    "delta": "0:00:00.003239",
    "end": "2022-03-11 13:10:14.394920",
    "item": "ip link add LSR-TST-br31 type veth peer name peerLSR-TST-br31",
    "rc": 255,
    "start": "2022-03-11 13:10:14.391681"
}
STDERR:
Error: argument "peerLSR-TST-br31" is wrong: "name" not a valid ifname
MSG:
non-zero return code
failed: [/home/nhosoi/.cache/linux-system-roles/rhel-8.qcow2] (item=ip link set peerLSR-TST-br31 up) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": [
        "ip",
        "link",
        "set",
        "peerLSR-TST-br31",
        "up"
    ],
    "delta": "0:00:00.003439",
    "end": "2022-03-11 13:10:14.769660",
    "item": "ip link set peerLSR-TST-br31 up",
    "rc": 1,
    "start": "2022-03-11 13:10:14.766221"
}
STDERR:
Cannot find device "peerLSR-TST-br31"
MSG:
non-zero return code
failed: [/home/nhosoi/.cache/linux-system-roles/rhel-8.qcow2] (item=ip link set LSR-TST-br31 up) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": [
        "ip",
        "link",
        "set",
        "LSR-TST-br31",
        "up"
    ],
    "delta": "0:00:00.006534",
    "end": "2022-03-11 13:10:15.149225",
    "item": "ip link set LSR-TST-br31 up",
    "rc": 1,
    "start": "2022-03-11 13:10:15.142691"
}
STDERR:
Cannot find device "LSR-TST-br31"
MSG:
non-zero return code
```
`interface: LSR-TST-br31` is defined and set using `set_fact` in `tests_bridge_nm.yml`, which intervenes `tests_change_indication_on_repeat_run.yml`.